pulsar/consumer_multitopic.go

```diff
@@ -167,6 +167,47 @@ func (c *multiTopicConsumer) AckID(msgID MessageID) error {
 	return mid.consumer.AckID(msgID)
 }
 
+func (c *multiTopicConsumer) AckIDList(msgIDs []MessageID) error {
+	return ackIDListFromMultiTopics(c.log, msgIDs, func(msgID MessageID) (acker, error) {
+		if !checkMessageIDType(msgID) {
+			return nil, errors.New("invalid message id type %T")
+		}
+		if mid := toTrackingMessageID(msgID); mid != nil && mid.consumer != nil {
+			return mid.consumer, nil
+		}
+		return nil, errors.New("consumer is nil")
+	})
+}
+
+func ackIDListFromMultiTopics(log log.Logger, msgIDs []MessageID, findConsumer func(MessageID) (acker, error)) error {
+	consumerToMsgIDs := make(map[acker][]MessageID)
+	for _, msgID := range msgIDs {
+		if consumer, err := findConsumer(msgID); err == nil {
+			consumerToMsgIDs[consumer] = append(consumerToMsgIDs[consumer], msgID)
+		} else {
+			log.Warnf("Can not find consumer for %v", msgID)
```

Comment:
Do we need to include this type of error in `AckError` as well?

Reply:
https://github.com/apache/pulsar-client-go/blob/6ffc7d283fbdb43e11d16bd643a7f28544cd9d1c/pulsar/consumer_impl.go#L564-L571

For example, if checkMsgIDPartition return an error, we need to tell the user.

Reply:
But users can do nothing to handle invalid message IDs except for excluding them from the next `AckIDList` call.

It will make users hard to handle the error.

Reply:
Ok, Make sense to me. The user can from warn log to get this.